src/hazelweb/model/ProgramEvaluator.rei

```diff
@@ -0,0 +1,193 @@
+open Lwtutil;
```

Comment:
could use some module-level documentation describing the architecture here, easy to get lost in the details below